src/Factory/OAuth2ServerFactory.php

```diff
@@ -88,6 +88,12 @@ private static function createStorageFromAdapter($adapter, array $config, Servic
             case 'mongo':
                 return self::createMongoAdapter($config, $services);
             default:
+                if ($services->has($adapter)) {
```

Comment:
I had this idea too.  It's not going to fly.  For a working implementation of a custom oauth2 adapter see https://github.com/API-Skeletons/zf-oauth2-doctrine